src/chat.rs

```diff
@@ -1907,6 +1908,54 @@ pub fn forward_msgs(context: &Context, msg_ids: &[MsgId], chat_id: u32) -> Resul
     Ok(())
 }
 
+pub fn get_info_json(context: &Context, chat_id: u32) -> Result<String, Error> {
+    let chat = Chat::load_from_db(context, chat_id).unwrap();
+
+    // ToDo:
+    // - [x] id
+    // - [x] type
+    // - [x] name
+    // - [x] archived
+    // - [x] color
+    // - [x] profileImage
+    // - [x] subtitle
+    // - [x] draft,
+    // - [ ] deaddrop,
+    // - [ ] summary,
+    // - [ ] lastUpdated,
+    // - [ ] freshMessageCounter,
+    // - [ ] email
+
+    let profile_image = match chat.get_profile_image(context) {
+        Some(path) => path.into_os_string().into_string().unwrap(),
+        None => "".to_string(),
+    };
+
+    let draft = match get_draft(context, chat_id) {
+        Ok(message) => match message {
+            Some(m) => m.text.unwrap_or("".to_string()),
```

Comment:
If calling a function I was told to use `.unwrap_or_else()` because otherwise the function is called regardless as this isn't lazy.

Reply:
Don't get it. Which function is called regardless?

Reply:
`to_string` is called regardless of whether `unwrap` fails or not. Clippy complains because `to_string` allocates memory even if there is no `unwrap` error.